pilot/pkg/model/context.go

```diff
@@ -600,6 +600,10 @@ const (
 
 	// NodeMetadataTLSClientRootCert is the absolute path to client root cert file
 	NodeMetadataTLSClientRootCert = "TLS_CLIENT_ROOT_CERT"
+
+	// NodeMetadataIdleTimeout specifies the idle timeout for the proxy, in duration format (10s).
+	// If not set, no timeout is set.
+	NodeMetadataIdleTimeout = "IDLE_TIMEOUT"
```

Comment:
FWIW, i think https://github.com/istio/api/pull/879 is the way to go, for this specific option. EnvoyFilter is a way to customize certain proxies. There is no point if everyone is using it to set the same global setting like idle timeout for all proxies. We should make such common cases easier for end users, instead of using break glass config like envoy filter.

We should get https://github.com/istio/api/pull/879 in, and use the current PR as a way of customizing the value for individual nodes. This is the same pattern we follow for all SDS fields, TLS cert paths, etc.

Reply:
If we are not okay with 879, then this should be a temp pilot env var like ISTIO11_IDLE_TIMEOUT so that we can get rid of it in 1.2.